docs/design/diagram-types/DesignDiagramsTemplates/classDiagrams.md

```diff
@@ -0,0 +1,74 @@
+# Class Diagrams
+
+## Purpose
+
+This document is intended to provide a baseline understanding for what, why, and how to incorporate Class Diagrams as part of your engagement. Regarding the **how**, the section at the bottom will provide tools and plugins to automate as much as possible when generating Class Diagrams through VSCode.
+
+[Wikipedia](https://en.wikipedia.org/wiki/Class_diagram) defines UML Class Diagrams as:
+
+ > _a type of static structure diagram that describes the structure of a system by showing the system's classes, their attributes, operations (or methods), and the relationships among objects._
+
+The key terms to make a note of here are:
+
+- static structure
+- showing the system's classes, attributes, operations, and relationships
+
+Class Diagrams are a type of a static structure because it focuses on the properties, and relationships of classes. It is not supposed to inform about the data flow, the caller or callee responsibilities, the request flows, nor any other "behavior" related characteristics.
+
+## Essential Takeaways
+
+Each "Component" (Stand alone piece of software - think datastores, microservices, serverless functions, user interfaces, etc...) of a Product or System will have it's own Class Diagram.
+
+Class Diagrams should tell a "story", where each Diagram will require Engineers to really think about:
+
+- The responsibility / operations of each class. What can (should) the class perform?
+- The class' attributes and properties. What can be set by an implementor of this class? What are all (if any) universally static properties?
+- The visibility or accessibility that a class' operation may have to other classes
+- The [relationship](https://en.wikipedia.org/wiki/Class_diagram#Relationships) between each class or the various instances
+
+## When to Create?
+
+Because Class Diagrams represent one of the more granular depiction of what a "product" or "system"
+is composed of, it is recommended to begin the creation of these diagrams at the beginning and throughout the engineering portions of an engagement.
+
+This does mean that any code change (new feature, enhancement, code refactor) might involve updating one or many Class Diagrams. Although this might seem like a downside of Class Diagrams, it actually can become a very strong benefit.
+
+Because Class Diagrams tell a "story" for each Component of a product (see the previous section), it requires a
+substantial amount of upfront thought and design considerations. This amount of upfront thought ultimately results in making more effective code changes, and may even minimize the level of refactors in future stages of the engagement.
```

Comment:
Looks like there are some line breaks in the md files - this makes it read funky (in md editors) depending on resolutions, so I added some suggestions to remove the line breaks, but to avoid repeating myself, I won't do it for the rest of the documents

```suggestion
Because Class Diagrams tell a "story" for each Component of a product (see the previous section), it requires a substantial amount of upfront thought and design considerations. This amount of upfront thought ultimately results in making more effective code changes, and may even minimize the level of refactors in future stages of the engagement.
```